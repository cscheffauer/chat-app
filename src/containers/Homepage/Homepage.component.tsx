import React, { useState } from 'react'
import LogIn from '../../components/LogIn/LogIn.component';
import ChatContainer from '../ChatContainer/ChatContainer.component';
import Header from '../../components/Header/Header.component';

import './Homepage.scss';

type Props = {
    client: any
}


const Homepage = ({ client }: Props) => {
    const [loggedIn, setloggedIn] = useState(false)
    const [username, setusername] = useState("" as String);

    const logInUser = (username: String) => {
        setusername(username);
        client.send(JSON.stringify({
            username,
            type: "userevent"
        }));

        setloggedIn(true);
    };


    return (
        <div className="homepage">
            {
                !loggedIn ?
                    <LogIn logInUser={logInUser} />
                    :
                    <ChatContainer client={client} username={username} />
            }
        </div>
    )
}

export default Homepage
