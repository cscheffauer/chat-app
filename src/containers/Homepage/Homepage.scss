@import '../../variables';

.homepage {
	width: 100%;
	height: 92vh; /* Fallback for browsers that do not support Custom Properties */
	height: calc(var(--vh, 1vh) * 92);
	@media only screen and (min-width: $chatwindowWidth) {
		width: $chatwindowWidth;
		position: absolute;
		left: calc(50% - #{$chatwindowWidth} / 2);
	}
}
