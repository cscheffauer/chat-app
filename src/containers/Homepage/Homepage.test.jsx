import React from 'react';
import { mount } from 'enzyme';
import Homepage from './Homepage.component';

describe('<Homepage />', () => {
	let wrapper;

	const clientMock = jest.fn();

	beforeEach(() => {
		wrapper = mount(<Homepage client={clientMock} />);
	});

	afterEach(() => {
		jest.clearAllMocks();
	});
	it('expect to render Homepage component', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
