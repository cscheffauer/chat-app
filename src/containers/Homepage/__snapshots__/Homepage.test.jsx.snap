// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Homepage /> expect to render Homepage component 1`] = `
<Homepage
  client={[MockFunction]}
>
  <div
    className="homepage"
  >
    <LogIn
      logInUser={[Function]}
    >
      <div
        className="login"
      >
        <p>
          Please type in your name to join:
        </p>
        <form
          onSubmit={[Function]}
        >
          <input
            onChange={[Function]}
          />
          <button
            type="submit"
          >
            Join
          </button>
        </form>
      </div>
    </LogIn>
  </div>
</Homepage>
`;
